internal/clients/team5/iito.go

```diff
@@ -6,49 +6,71 @@ import "github.com/SOMAS2020/SOMAS2020/internal/common/shared"
-		- allows client to notify server of all gift responses we received. We would need to store these
-		transactions to be able to inform server when it calls this method
+	DecideGiftAmount(toTeam shared.ClientID, giftOffer shared.Resources) shared.Resources
 	SentGift(sent shared.Resources, to shared.ClientID)
-		- notifies us that gift was successfully sent. Can use this as a time to check resource deduction.
 	ReceivedGift(sent shared.Resources, from shared.ClientID)
-		- notifies us that gift was successfully received. Can use this as a time to check resource addition.
-	DecideGiftAmount(toTeam shared.ClientID, giftOffer shared.Resources) shared.Resources
-		- called by server at end of round. Allows us to choose how much of our gift offers we actually want to fulfill.
+
+	For refrence
+	OurRequest giftInfo
+		requested      shared.GiftRequest    - GetGiftRequest - Amount WE request 1
+		offered        shared.GiftOffer 	- GetGiftResponses - Amount offered TO US 3
+		response       shared.GiftResponse	- GetGiftResponses - Amount WE accepeted and reason 3
+		actualRecieved shared.Resources		- ReceivedGift - Amount WE actually received 6
+
+	TheirRequest giftInfo
+		requested      shared.GiftRequest    - GetGiftOffers - Amount THEY requested  2
+		offered        shared.GiftOffer		- GetGiftOffers - Amount WE offered 2
+		response       shared.GiftResponse 	- UpdateGiftInfo - Amount THEY accepeted and reason 4
+		actualRecieved shared.Resources		- DecideGiftAmount - Amount THEY actually get 5
 */
 
+// creates initial opinions of clients and sets the values to 0 (prevents nil mapping)
+func (c *client) initGiftHist() {
+	c.giftHistory = giftHistory{}
+
+	for _, team := range c.getAliveTeams(true) {
+		ourGiftInfo := giftInfo{
+			requested:      0,
+			offered:        0,
+			response:       shared.GiftResponse{AcceptedAmount: 0, Reason: 0},
+			actualRecieved: 0,
+		}
+		theirGiftInfo := giftInfo{
+			requested:      0,
+			offered:        0,
+			response:       shared.GiftResponse{AcceptedAmount: 0, Reason: 0},
+			actualRecieved: 0,
+		}
+		ourReq := map[uint]giftInfo{c.gameState().Turn: ourGiftInfo}
+		theirReq := map[uint]giftInfo{c.gameState().Turn: theirGiftInfo}
+		c.giftHistory[team] = giftExchange{
+			OurRequest:   ourReq,
+			TheirRequest: theirReq,
+		}
+	}
+}
+
 // GetGiftRequests we want gifts!
 func (c *client) GetGiftRequests() shared.GiftRequestDict {
 	requests := shared.GiftRequestDict{}
-	switch {
-	case c.gameState().ClientInfo.LifeStatus == shared.Critical: // Case we are critical
-		for team, status := range c.ServerReadHandle.GetGameState().ClientLifeStatuses {
-			if status == shared.Critical { // Other island are critical
-				requests[team] = shared.GiftRequest(0.0) // Dont ask for money
-			} else {
+	for team, status := range c.gameState().ClientLifeStatuses {
+		if status != shared.Critical {
+			switch {
+			case c.gameState().ClientInfo.LifeStatus == shared.Critical: // Case we are critical
```

Comment:
Also a helper method to get life status: `getLifeStatus()` - would prefer if you use that. Pull in latest from develop if you don't have it